pkg/doc/verifiable/credential.go

```diff
@@ -452,90 +436,121 @@ func decodeCredentialSchema(data []byte) ([]TypedID, error) {
 	return nil, errors.New("verifiable credential schema of unsupported format")
 }
 
-// NewCredential creates an instance of Verifiable Credential by reading a JSON document from bytes.
-// It also applies miscellaneous options like custom decoders or settings of schema validation.
-func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, error) {
-	// Apply options
-	crOpts := defaultCredentialOpts()
-
-	for _, opt := range opts {
-		opt(crOpts)
-	}
+// NewCredential decodes Verifiable Credential from bytes which could be marshalled JSON or serialized JWT.
+// It also applies miscellaneous options like settings of schema validation.
+// It returns decoded Credential and its marshalled JSON.
+// For JSON bytes input, the output marshalled JSON is the same value.
+// For serialized JWT input, the output is the result of decoding `vc` claim from JWT.
+// The output Credential and marshalled JSON can be used for extensions of the base data model
```

Comment:
At first, I thought that this second return value might be used as an input to NewExtCredential. But that doesn't make sense since NewExtCredential calls NewCredential itself.

So I think we could potentially create confusion with the current pattern.

Reply:
I agree it is bit confusing